src/logger/BaseLogger.go

```diff
@@ -0,0 +1,144 @@
+package logger
+
+//this doesn't resolve, don't know why
+//not super important right now
+// import (
+// 	"net/url"
+// )
+
+/**
+ * Initialize enabled logger using default url.
+ */
+func NewBaseLoggerAgent(_agent string) *BaseLogger {
+	return NewBaseLogger(_agent, "insert default url here", true, nil)
```

Comment:
I think we can assume this is coming from a UseLogger singleton we will have to contruct